jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushToRepositoryAction.kt

```diff
@@ -0,0 +1,360 @@
+                fromDockerfileButton = this@row.radioButton(message("ecr.push.type.dockerfile.label"), BuildType.Dockerfile).component
+            }
+        }
+
+        val imageSelectorPanel = localImageSelectorPanel()
+        val dockerfilePanel = dockerfileConfigurationSelectorPanel()
+
+        row {
+            cell(isFullWidth = true) {
+                imageSelectorPanel(grow)
+                    .installOnParent { fromLocalImageButton.isSelected }
+                    .visibleIf(fromLocalImageButton.selected)
+                dockerfilePanel(grow)
+                    .installOnParent { fromDockerfileButton.isSelected }
+                    .visibleIf(fromDockerfileButton.selected)
+            }
+        }
+
+        row(message("ecr.repo.label")) {
+            component(remoteRepos)
+                .growPolicy(GrowPolicy.MEDIUM_TEXT)
+                .withErrorOnApplyIf(message("ecr.repo.not_selected")) { it.selected() == null }
+        }
+
+        row(message("ecr.push.remoteTag")) {
+            textField(::remoteTag)
+                .constraints(grow)
+                .also {
+                    it.component.emptyText.text = defaultTag
+                }
+        }
+    }
+
+    private fun localImageSelectorPanel() = panel {
+        row(message("ecr.push.source")) {
+            // property binding syntax causes kotlin compiler error for some reason
+            comboBox(
+                localImageRepoTags,
+                { localImage },
+                { ::localImage.set(it) },
+                listCellRenderer { value, _, _ ->
+                    text = value.tag ?: value.imageId.take(15)
+                }
+            )
+                .growPolicy(GrowPolicy.MEDIUM_TEXT)
+                .withErrorOnApplyIf(message("ecr.image.not_selected")) { it.selected() == null }
+        }
+    }
+
+    private fun dockerfileConfigurationSelectorPanel() = panel {
+        row(message("ecr.dockerfile.configuration.label")) {
+            cell {
+                val model = CollectionComboBoxModel<DockerRunConfiguration>()
+                rebuildRunConfigurationComboBoxModel(model)
+                val box = comboBox(
+                    model,
+                    { runConfiguration },
+                    { ::runConfiguration.set(it) },
+                    listCellRenderer { value, _, _ ->
+                        icon = value.icon
+                        text = value.name
+                    }
+                )
+                    .growPolicy(GrowPolicy.MEDIUM_TEXT)
+                    .withErrorOnApplyIf(message("ecr.dockerfile.configuration.invalid")) { it.selected() == null }
+                    .withErrorOnApplyIf(message("ecr.dockerfile.configuration.invalid_server")) { it.selected()?.serverName == null }
+
+                // TODO: how do we render both the Docker icon and action items correctly?
```

Comment:
This todo was fulfilled right? at least it looked correct when I tried it

Reply:
![image](https://user-images.githubusercontent.com/742829/116468332-9448fe00-a825-11eb-9b16-f9309689b2d9.png)

my goal was to have the Docker icon show up, but current state feels good enough for now